sklearn/_isotonic.pyx

```diff
@@ -75,10 +76,10 @@ def _make_unique(cnp.ndarray[dtype=floating] X,
     """
     unique_values = len(np.unique(X))
 
-    cdef cnp.ndarray[dtype=floating] y_out = np.empty(unique_values,
+    cdef const floating[::1] y_out = np.empty(unique_values,
```

Comment:
As noted in the [CI error](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=53192&view=logs&j=dde5042c-7464-5d47-9507-31bdd2ee0a3a&t=7d852497-2547-55fa-986f-0b436c028d7e): the `*_out` variables need to be non-`const` because `_make_unique` itself is modifying them.